src/index.ts

```diff
@@ -29,75 +29,84 @@ function monitorConnections() {
 	});
 }
 
-app.get(`/health`, (req, res) => {
-	if (terminated) {
-		logger.info('health check failed due to application terminating');
-
-		return res.status(429).send('terminated');
+const setupServer = async () => {
+	if (process.env.NODE_ENV !== 'production') {
+		const { default: setupDev } = await import('./setupDev');
+		await setupDev(app);
 	}
 
-	return res.status(200).send('ok');
-});
+	app.get(`/health`, (req, res) => {
+		if (terminated) {
+			logger.info('health check failed due to application terminating');
 
-app.use(router);
-initGraphql(app);
+			return res.status(429).send('terminated');
```

Comment:
for some reason functional tests in github seem to fail.. they rely on this endpoint I believe 🤔

Reply:
Yes, I'm not able to reproduce it in my local env, even with tools like [act](https://github.com/nektos/act). Could we see the container's logs to see what's happening? Because I think they're not being thrown to the output, so I'm going in a little blind on this 😅

I've added some changes to check if the problem is with the NODE_ENV value when executing the app during the tests, but I don't know if it will fix it 🤔